Libplanet/Blockchain/CanonicalChainComparer.cs

```diff
@@ -0,0 +1,33 @@
+#nullable enable
+using System.Collections.Generic;
+using Libplanet.Blockchain.Policies;
+using Libplanet.Blocks;
+
+namespace Libplanet.Blockchain
+{
+    /// <summary>
+    /// The default canonical chain comparer (which purpose to be a <see
+    /// cref="IBlockPolicy{T}.CanonicalChainComparer"/>).
+    /// <para>The chain which has the most <see cref="Block{T}.TotalDifficulty"/> is considered
+    /// the greatest, i.e., canonical chain.</para>
+    /// </summary>
+    /// <seealso cref="IBlockPolicy{T}.CanonicalChainComparer"/>
+    /// <seealso cref="IBlockExcerpt"/>
+    public class CanonicalChainComparer : IComparer<IBlockExcerpt>
```

Comment:
IMHO, `CanonicalChainComparer` doesn't represent its behavior... how about a name like `DifficultyBasedChainComparer`?

Reply:
The logic behind this will be changed soon.  How about `DefaultCanonicalChainComparer`, to imply it is the officially most recommended configuration?

Reply:
There could be multiple implementation of comparer, so to clarify its usage `DifficultyBasedChainComparer` seems better to me. To suggest it as default, maybe write it as docstring will help I guess?

Reply:
If we put the implementation details into the name, we should rename it when we change the details.  This can be considered as a kind of leaky abstraction in theory, and also makes difficult to maintain backward compatibility of API due to the name: we should keep the name even if it no more reflects the implementation, or rename it to break backward compatibility.  I believe this would not be good for the default implementation. (It'd be good if C# supports class aliases like `public class DefaultChainComparer = DifficultyBasedChainComparer`, but it's impossible anyway…)

Reply:
Or we can make `DefaultChainComparer` to subclass `DifficultyBasedChainComparer`.  How about it?

Reply:
IMO, if we want to represent this comparer as default, we can also consider adding static field `BlockPolicy.DefaultChainCoamparer` and setting it `new DifficultyBasedChainComparer()`.

Reply:
I renamed it to `TotalDifficultyComparer`: 9c5063451c1e012c2df804ebc733481017194203.